lib/common.sh

```diff
@@ -152,24 +152,38 @@ resolve_ipv4()
     local hname="$1"
 
     # Resolve hostname to IPv4 address.
-    host_op=$(host -4 -t A "$hname" | sort) 
+    host_op=$(host -4 -t A "$hname" | sort | shuf --random-source=/etc/machine-id)
```

Comment:
we cannot use /etc/machine-id directly. It may not be present on some machine.
Instead what I'm thinking is we should create a file at install time,

uuidgen > /opt/microsoft/aznfs/machine-id

and then use that file here. That way we are guaranteed that we have a (uniq) file to use for shuffling.
installer can make this file immutable.
uuidgen should mostly be present, if not it can install it first.

also, we can filter out the IP addresses right here and then use $host_op directly in the following code, that way shuffling is more appropriate. So this becomes

host_op=$(host -4 -t A "$hname" | grep " has address " | awk '{print $4}' | sort | shuf --random-source=$OPTDIR/machine-id)

also explain the rationale behind shuffling in the comment.

ALSO NOTE THAT YOU CANNOT DIRECTLY DO THE ABOVE. The host command has to be all by itself, w/o even the sort, o/w we cannot check the return status at L156. So you will have to run the host command by itself and then check the status and if that succeeds run a second command as above, f.e..

host_op=$(host -4 -t A "$hname")

now we can check the return status

now

host_op=$(echo "$host_op" | grep " has address " | awk '{print $4}' | sort | shuf --random-source=$OPTDIR/machine-id)